R/zzz.R

```diff
@@ -1,16 +1,20 @@
-
 tabpfn <- NULL
 
 .onLoad <- function(...) {
+  pkg <- asNamespace("TabPFN")
+  pkg$check_libomp()
 
   reticulate::py_require("tabpfn")
 
   tryCatch(
-    tabpfn <<- reticulate::import("tabpfn", delay_load = list(
-      on_error = function(e) {
-        cli::cli_abort(msg_tabpfn_not_available(e))
-      }
-    )),
+    tabpfn <<- reticulate::import(
+      "tabpfn",
+      delay_load = list(
```

Comment:
A good place to call `check_libomp()` is from a `before_load = function() { ... }` hook, (`before_load` as peer to `on_error`)

Reply:
`before_load()` as a (future) argument to `reticulate::import()`?

Reply:
nope, supported today

Reply:
See "Delay Load" section https://rstudio.github.io/reticulate/dev/reference/import.html